packages/beacon-state-transition/src/block/processExecutionPayload.ts

```diff
@@ -1,13 +1,14 @@
-import {bellatrix, ssz} from "@chainsafe/lodestar-types";
+import {bellatrix, ssz, allForks} from "@chainsafe/lodestar-types";
 import {toHexString, byteArrayEquals} from "@chainsafe/ssz";
 import {CachedBeaconStateBellatrix} from "../types.js";
 import {getRandaoMix} from "../util/index.js";
 import {ExecutionEngine} from "../util/executionEngine.js";
 import {isMergeTransitionComplete} from "../util/bellatrix.js";
 
-export function processExecutionPayload(
+export function processExecutionPayload<T extends allForks.BlockType>(
+  type: T,
```

Comment:
If would be nice if we can avoid having to pass this type everywhere

Reply:
changed, now conditioned on the body's execution payload/header inspection of the block :+1: